addons/crm/models/crm_lead.py

```diff
@@ -551,6 +552,12 @@ def _compute_partner_phone_update(self):
         for lead in self:
             lead.partner_phone_update = lead._get_partner_phone_update()
 
+    @api.depends_context('uid')
+    @api.depends('partner_id')
+    def _compute_is_partner_visible(self):
+        for lead in self:
+            lead.is_partner_visible = bool(lead.partner_id or lead.user_has_groups('base.group_no_one'))
```

Comment:
Would be a bit better if you assigned the group to a "is_debug_mode" variable before looping.

That way you would use "self.user_has_groups(...)" instead of using "lead.user_has_groups", which is a bit confusing since that check is in no way related to that record (I know it still works, but it makes it strange).

Also, we should always set this value to True for opportunities.
So the check would become `lead.type == 'opportunity' or lead.partner_id or is_debug_mode`.
(Meaning you need to add "type" to the depends).